//
// Copyright (C) 2024 EA group inc.
// Author: Jeff.li lijippy@163.com
// All rights reserved.
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published
// by the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
//


syntax="proto2";
package test;

option cc_generic_services = true;

message NotifyRequest {
      required string message = 1;
};

message NotifyResponse {
      required string message = 1;
};

enum ChangeType {
    FLUCTUATE = 1;  // Fluctuating between upper and lower bound 
    SMOOTH = 2;     // Smoothly rising from the lower bound to the upper bound 
}

message Stage {
    required int32 lower_bound = 1;
    required int32 upper_bound = 2;
    required int32 duration_sec = 3;
    required ChangeType type = 4; 
}

message TestCase {
    required string case_name = 1;
    required string max_concurrency = 2;
    repeated Stage qps_stage_list = 3;
    repeated Stage latency_stage_list = 4;
}

message TestCaseSet {
    repeated TestCase test_case = 1;
}

service ControlService {
    rpc Notify(NotifyRequest) returns (NotifyResponse);
}

service EchoService {
    rpc Echo(NotifyRequest) returns (NotifyResponse);
};

